test/contracts/kill_switch/kill_switch.test.js

```diff
@@ -0,0 +1,963 @@
+        await acl.createPermission(owner, killSwitch.address, SET_ALLOWED_INSTANCES_ROLE, root, { from: root })
+        await acl.createPermission(owner, killSwitch.address, SET_DENIED_BASE_IMPLS_ROLE, root, { from: root })
+        await acl.createPermission(owner, killSwitch.address, SET_HIGHEST_ALLOWED_SEVERITY_ROLE, root, { from: root })
+      })
+
+      beforeEach('create kill switched app', async () => {
+        const receipt = await dao.newAppInstance(SAMPLE_APP_ID, appBase.address, '0x', false, { from: root })
+        app = KillSwitchedApp.at(getNewProxyAddress(receipt))
+        await app.initialize(owner)
+      })
+
+      describe('integration', () => {
+        const itExecutesTheCall = () => {
+          it('executes the call', async () => {
+            await app.write(10, { from: owner })
+            assert.equal(await app.read(), 10)
+          })
+        }
+
+        context('when the function being called is not tagged', () => {
+          itExecutesTheCall()
+        })
+
+        context('when the function being called is tagged', () => {
+          const itAlwaysExecutesTheCall = () => {
+            context('when the instance being called is allowed', () => {
+              beforeEach('allow instance', async () => {
+                await killSwitch.setAllowedInstance(app.address, true, { from: owner })
+              })
+
+              context('when the base implementation is not denied', () => {
+                beforeEach('do not deny base implementation', async () => {
+                  await killSwitch.setDeniedBaseImplementation(appBase.address, false, { from: owner })
+                })
+
+                itExecutesTheCall()
+              })
+
+              context('when the base implementation is denied', () => {
+                beforeEach('deny base implementation', async () => {
+                  await killSwitch.setDeniedBaseImplementation(appBase.address, true, { from: owner })
+                })
+
+                itExecutesTheCall()
+              })
+            })
+
+            context('when the instance being called is not marked as allowed', () => {
+              beforeEach('dot not allow instance', async () => {
+                await killSwitch.setAllowedInstance(app.address, false, { from: owner })
+              })
+
+              context('when the base implementation is not denied', () => {
+                beforeEach('do not deny base implementation', async () => {
+                  await killSwitch.setDeniedBaseImplementation(appBase.address, false, { from: owner })
+                })
+
+                itExecutesTheCall()
+              })
+
+              context('when the base implementation is denied', () => {
+                beforeEach('deny base implementation', async () => {
+                  await killSwitch.setDeniedBaseImplementation(appBase.address, true, { from: owner })
+                })
+
+                itExecutesTheCall()
```

Comment:
Shouldn't this revert?

Reply:
Note that the `context` for this test is `"when the kernel is initialized with a failing kill-switch implementation"`. We decided that if the kill-switch was failing then we will allow calling the app being queried, it would be the same case as an "old-version" kernel

Reply:
:+1: Ahh yes, missed that this was under the "failing kill-switch" context!